equinox/nn/_conv.py

```diff
@@ -134,21 +174,34 @@ def __init__(
         self.out_channels = out_channels
         self.kernel_size = kernel_size
         self.stride = stride
-        if isinstance(padding, int):
-            self.padding = tuple((padding, padding) for _ in range(num_spatial_dims))
-        elif isinstance(padding, Sequence) and len(padding) == num_spatial_dims:
-            if all_sequences(padding):
-                self.padding = tuple(padding)
-            else:
-                self.padding = tuple((p, p) for p in padding)
-        else:
-            raise ValueError(
-                "`padding` must either be an int or tuple of length "
-                f"{num_spatial_dims} containing ints or tuples of length 2."
-            )
+        self.padding = _padding_init(padding, num_spatial_dims)
         self.dilation = dilation
         self.groups = groups
         self.use_bias = use_bias
+        self.padding_mode = _padding_mode_init(padding_mode)
+
+    def _nonzero_pad(self, x: Array) -> Array:
+        if isinstance(self.padding, str):
+            rhs_shape = tuple(
+                d * (k - 1) + 1 for k, d in zip(self.kernel_size, self.dilation)
+            )
+            padding = lax.padtype_to_pads(
```

Comment:
I don't think this is public API, unfortunately. ([It's not listed here, anyway.](https://jax.readthedocs.io/en/latest/jax.lax.html)) Maybe we can ask to add it to the public API for JAX?

Reply:
Could you raise a new issue in JAX? I think your proposal is much more likely to be accepted

Reply:
Done! We'll see what they say.